compiler/rustc_middle/src/ty/closure.rs

```diff
@@ -238,10 +234,14 @@ impl<'tcx> CapturedPlace<'tcx> {
     }
 }
 
-fn symbols_for_closure_captures(tcx: TyCtxt<'_>, def_id: (LocalDefId, LocalDefId)) -> Vec<Symbol> {
-    let typeck_results = tcx.typeck(def_id.0);
-    let captures = typeck_results.closure_min_captures_flattened(def_id.1);
-    captures.into_iter().map(|captured_place| captured_place.to_symbol(tcx)).collect()
+fn closure_captures<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    def: LocalDefId,
+) -> &'tcx [&'tcx ty::CapturedPlace<'tcx>] {
+    let (DefKind::Closure | DefKind::Generator) = tcx.def_kind(def) else { return &[] };
```

Comment:
is this ever called for something other than closures or generators? Could just ICE here and adjust the call sites to reduce the query calls

Reply:
nevermind 😆 done in a later commit